README.md
# socketsProject
Networking project


History.txt
2021-07-06 18:45:30.396762 : A.txt ('127.0.0.1', 20406)
2021-07-06 18:45:36.313415 : A.txt ('127.0.0.1', 20407)
2021-07-06 19:20:49.307403 : A.txt ('127.0.0.1', 26264)
2021-07-06 19:23:52.385602 : B.txt ('127.0.0.1', 19018)
2021-07-06 19:25:09.355005 : B.txt ('127.0.0.1', 2528)
2021-07-06 19:25:58.714630 : A.txt ('127.0.0.1', 1052)
2021-07-06 19:54:44.094843 : A.txt ('127.0.0.1', 6761)
2021-07-06 19:56:08.659566 : c.txt ('127.0.0.1', 3454)
2021-07-06 19:58:16.881592 : A.txt ('127.0.0.1', 5204)
2021-07-06 20:04:40.635503 : B.txt ('127.0.0.1', 6006)
2021-07-06 20:05:00.857634 : B.txt ('127.0.0.1', 1048)
2021-07-06 20:21:30.730679 : A.txt ('127.0.0.1', 1054)
2021-07-06 20:28:52.782797 : A.txt ('127.0.0.1', 1056)
2021-07-06 20:30:56.575436 : A.txt ('127.0.0.1', 31158)
2021-07-06 20:36:18.397511 : A.txt ('127.0.0.1', 1102)
2021-07-06 20:38:00.432398 : A.txt ('127.0.0.1', 29963)
2021-07-06 21:12:32.629879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.633878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.634878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.636878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.638882 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.639880 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.641900 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.643878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.645881 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.647879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.649879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.650878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.652879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.653881 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.655878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.657879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.659891 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.661876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.663897 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.665877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.666876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.668878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.670880 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.671879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.673879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.676916 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.679879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.680877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.682876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.684876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.685875 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.687877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.688880 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.690878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.691890 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.693880 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.695877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.697881 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.699878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.701879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.702878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.704876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.705884 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.707877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.708892 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.710878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.711876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.714875 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.716876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.717892 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.719877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.720876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.722878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.723876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.725876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.726884 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.728878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.730876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.731876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.733878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.734876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.736877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.737877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.739876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.740876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.742877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.743876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.745877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.746877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.748876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.750877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.751876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.753876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.754878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.756877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.757877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.759878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.760876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.762878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.764877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.765878 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.767894 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.768876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.769877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.771875 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.772890 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.773891 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.774875 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.776879 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.778877 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.780876 : A.txt ('127.0.0.1', 5675)
2021-07-06 21:12:32.781876 : A.txt ('127.0.0.1', 5675)

Client.py
import socket
from tkinter import *
import os


#Definition des varibales globales

filename = ""
BUFFER_SIZE = 4096
Host = "127.0.0.1"
Port = 5000

#connexion au serveur
s  = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((Host, Port))


#Definition de la fonction de déconnexion (boutton quitter)
    
def quit(root):
    global s
    try:
        s.send("Q".encode())
        s.close()
    except:
        reConnect()
        s.send("Q".encode())
        s.close()

    root.destroy()





#une fonction pour la reconnexion au cas d'erreur
def reConnect():
    global s
    s.close()
    s  = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.connect((Host, Port))


#la fonction de selection des fichiers: avecl'envoie du choix et la réception du fichier
def CurSelet(evt):
    global filename
    
    filename = str(listbox.get(listbox.curselection()))
    value = filename.encode()
    try:
        s.sendall(value)
    except:
        reConnect()
        s.sendall(value)
    
    with open(os.path.join("Client",filename.replace("\n", "")), "wb") as f:
        while True:
           
           
            bytes_read = s.recv(BUFFER_SIZE)
            if not bytes_read:    
                
                break
            else:
            
                f.write(bytes_read)
        
        f.close()
            


if __name__ == "__main__":
    
    

# definition de la fenêtre 

    root = Tk()
    root.title("Client")
    sizex = 600
    sizey = 400
    posx  = 40
    posy  = 20
    root.wm_geometry("%dx%d+%d+%d" % (sizex, sizey, posx, posy))
    label = Label(root, text = "Liste des fichiers")

    label.pack() 


#Ajout des fichiers disponibles (dans files.txt) à l'interface
    fileList = []
    file = open("files.txt", "r")
    lines = file.readlines()
    for l in lines:
        fileList.append(l)
    file.close()
    listbox = Listbox(root,width=60,height=10,font=('times',13))
    for f in fileList:
        listbox.insert(END, f)
    listbox.pack()
    listbox.bind('<Double-1>',CurSelet)

#ajouter un boutton de déconnexion
    button = Button(root, text="Quit", command=lambda root=root:quit(root))
    button.pack()



    
    
           


    
    
    root.mainloop()
    

Server2.py
import threading
from tkinter import *
import socket
from datetime import datetime
import os

#Définition des variables
SERVER_HOST = "0.0.0.0"
SERVER_PORT = 5000

BUFFER_SIZE = 4096
folderName = "./files"
fileName = "files.txt"


#Ecriture des noms des fichiers (dispo dans le dossier files) dans le fichier files.txt
def writeAll(folderName, fileName):
    FilesList = [f for f in os.listdir(folderName) if os.path.isfile(os.path.join(folderName, f))]
    with open(fileName, "w") as inp:
        for f in FilesList:
            inp.write(f+"\n")
        inp.close()


#L'application principale (Main thread)
class MyApp():
   
   def __init__(self, mywin):
        self.mywin = mywin
        
        writeAll(folderName, fileName)
        
            
#Définition de la fenêtre
root = Tk()
root.title("Server")
sizex = 600
sizey = 400
posx  = 40
posy  = 20
root.wm_geometry("%dx%d+%d+%d" % (sizex, sizey, posx, posy))
label = Label(root, text = "Liste des requêtes")



Label1 = Label(root, text="Date et Heure")  
Label2 = Label(root, text="Source") 
Label3 = Label(root, text="Nom de Fichier")
Label4 = Label(root, text="Status")


label.grid(row = 0, column = 1, sticky = W, padx = 2)

Label1.grid(row = 1, column = 1, sticky = W, padx = 2,pady=2)
Label2.grid(row = 1, column = 2, sticky = W, padx = 2,pady=2)
Label3.grid(row = 1, column = 3, sticky = W, padx = 2,pady=2) 
Label4.grid(row = 1, column = 4, sticky = W, padx = 2,pady=2) 

cpt = 2
myapp = MyApp(root)


#Le coté socket qui comprend :connexion, réception et envoie
def mainloop():
    global cpt
    
    while True:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        s.bind((SERVER_HOST, SERVER_PORT))
        s.listen()
        client_socket, address = s.accept()
        received = client_socket.recv(BUFFER_SIZE).decode()
        if(received == "Q"):
            break
        else:
            h = open("history.txt", "a")
            h.write(str(datetime.now())+" : "+str(received).replace("\n","")+" "+str(address)+"\n")
            h.close()
            
            Label1 = Label(root, text=str(datetime.now()))  
            Label3 = Label(root, text=str(received).replace("\n","")) 
            Label2 = Label(root, text=str(address[0])+":"+str(address[1]))

            Label1.grid(row = cpt, column = 1, sticky = W, padx = 2)
            Label2.grid(row = cpt, column = 2, sticky = W, padx = 2)
            Label3.grid(row = cpt, column = 3, sticky = W, padx = 2) 

            status = "Succes"
            try:
                with open(os.path.join(folderName,received.replace("\n","")), "rb") as f:
                    
                    while True:
                        
                        bytes_read = f.read(BUFFER_SIZE)
                        if not bytes_read:
                            
                            break
                        
                        client_socket.sendall(bytes_read)
                    f.close()
                    
            except:
                status = "Failed"
            Label4 = Label(root, text=status)
            Label4.grid(row = cpt, column = 4, sticky = W, padx = 2) 


            cpt+=1
            
        client_socket.close()

    s.close()

threading.Thread(target=mainloop).start()
root.mainloop()


Server.py
import socket
from tkinter import *
from tkinter.ttk import *
import os
import threading
from datetime import datetime

def writeAll(folderName, fileName):
    FilesList = [f for f in os.listdir(folderName) if os.path.isfile(os.path.join(folderName, f))]
    with open(fileName, "w") as inp:
        for f in FilesList:
            inp.write(f+"\n")
        inp.close()


root = Tk()
root.title("Server")
sizex = 600
sizey = 400
posx  = 40
posy  = 20
root.wm_geometry("%dx%d+%d+%d" % (sizex, sizey, posx, posy))
label = Label(root, text = "Liste des requêtes")



Label1 = Label(root, text="Date et Heure")  
Label2 = Label(root, text="Source") 
Label3 = Label(root, text="Nom de Fichier")
Label4 = Label(root, text="Status")


label.grid(row = 0, column = 1, sticky = W, padx = 2,pady=2)
label.place(x=25, y=25, anchor="center")
Label1.grid(row = 1, column = 1, sticky = W, padx = 2,pady=2)
Label2.grid(row = 1, column = 2, sticky = W, padx = 2,pady=2)
Label3.grid(row = 1, column = 3, sticky = W, padx = 2,pady=2) 
Label4.grid(row = 1, column = 4, sticky = W, padx = 2,pady=2) 

cpt = 2


SERVER_HOST = "0.0.0.0"
SERVER_PORT = 5000

BUFFER_SIZE = 4096
folderName = "./files"
fileName = "files.txt"
writeAll(folderName, fileName)
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind((SERVER_HOST, SERVER_PORT))
s.listen()
client_socket, address = s.accept()
root.mainloop()


received = client_socket.recv(BUFFER_SIZE).decode()


while(received != "Q"):
    h = open("history.txt", "a")
    h.write(str(datetime.now())+" : "+str(received).replace("\n","")+" "+str(address)+"\n")
    h.close()
    # print(received)
    # Label1 = Label(root, text=str(datetime.now()))  
    # Label3 = Label(root, text=str(received).replace("\n","")) 
    # Label2 = Label(root, text=str(address[0])+":"+str(address[1]))

    # Label1.grid(row = cpt, column = 1, sticky = W, padx = 2)
    # Label2.grid(row = cpt, column = 2, sticky = W, padx = 2)
    # Label3.grid(row = cpt, column = 3, sticky = W, padx = 2) 

    status = "Succes"
    try:
        with open(os.path.join(folderName,received.replace("\n","")), "rb") as f:
            
            while True:
                print("here")
                bytes_read = f.read(BUFFER_SIZE)
                if not bytes_read:
                    
                    break
                
                client_socket.sendall(bytes_read)
            f.close()
            print("done")
    except:
        status = "Failed"
    # Label4 = Label(root, text=status)
    # Label4.grid(row = cpt, column = 4, sticky = W, padx = 2) 


    # cpt+=1
    # received = client_socket.recv(BUFFER_SIZE).decode()
    
client_socket.close()

s.close()





